Checks/PWR003/gravity.f90

```diff
@@ -0,0 +1,42 @@
+module gravity_getters
+contains
+  ! Returns the gravity on Earth in m/s2
+  pure function gravity_pure(planetIndex) bind(c)
+    use iso_c_binding, only: c_int, c_double
+    
+    implicit none
+    integer(kind=c_int), intent(in) :: planetIndex
+    real(kind=c_double) :: gravity_pure
+
+    SELECT CASE (planetIndex) 
+      CASE (0) ! Mercury
+        gravity_pure = 3.7
+      CASE (1) ! Venus
+        gravity_pure = 8.9
+      CASE (2) ! Earth
+        gravity_pure = 9.8
+      CASE (3) ! Mars
+        gravity_pure = 3.7
+      CASE (4) ! Jupiter
+        gravity_pure = 23.1
+      CASE (5) ! Saturn
+        gravity_pure = 9.0
+      CASE (6) ! Uranus
+        gravity_pure = 8.7
+      CASE (7) ! Neptune
+        gravity_pure = 11.0
+      CASE DEFAULT
+        gravity_pure = 0.0
+    END SELECT
+  end function gravity_pure
+  
+  function gravity_f(planetIndex) bind(c)
+    use iso_c_binding, only: c_int, c_double
+    
+    implicit none
+    integer(kind=c_int), intent(in) :: planetIndex
+    real(kind=c_double) :: gravity_f
+    
+    gravity_f = gravity_pure(planetIndex)
+  end function gravity_f
```

Comment:
I don't think you should call `gravity_pure` from `gravity_f`. I believe `gravity_f` should be an exact copy of `gravity_pure`, but without `pure`. Also, I would rename `gravity_f` to `gravity_impure`.

Reply:
If we call `gravity_pure` from within `gravity_f`, we are doing 2 function calls, whereas in the `solution` we would making 1 function call only. We want the benchmark to show if `pure` helps the Fortran compiler here. We don't want to show the difference between two function calls versus one function call in a hot loop.

Reply:
I totally agree with you, during the testing that was my first approach, but since I got 0 boost performance, I did prefer to not duplicate code, but looking that as you say, it doesn't has sense.
Also I can perfectly rename `gravity_f` to `impure` as I just called `_f` because I had some linker troubles